pkg/api-server/resource_endpoints.go

```diff
@@ -360,6 +360,13 @@ func (r *resourceEndpoints) createResource(
 		return
 	}
 
+	if r.federatedZone && r.descriptor.IsPluginOriginated {
```

Comment:
should this be only in federated zone? Shouldn't we need this in non-federated zone and potentially global too?

Reply:
Ah right, a non-federated zone should have this webhook for sure. Also, we should probably add `kuma.io/origin` to all resources, not just plugin-originated policies.